modules/test/playwright/tests/frontend-taglib-clay/main/cards.spec.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+
+import {loginTest} from '../../../fixtures/loginTest';
+import {clickAndExpectToBeVisible} from '../../../utils/clickAndExpectToBeVisible';
+import {claySamplePageTest} from './fixtures/claySamplePageTest';
+import {TabName} from './pages/ClaySamplePage';
+
+const test = mergeTests(claySamplePageTest, loginTest());
+
+test.beforeEach(async ({claySamplePage}) => {
+	await claySamplePage.selectTab(TabName.CARDS);
+});
+
+test('Asserts selectable image cards behavior', async ({page}) => {
+	const selectableImageCardsSection = page.locator(
+		'div.h4:has-text("Selectable Image Card") + div.row'
+	);
+
+	const selectableImageCards =
+		selectableImageCardsSection.locator('.form-check-card');
+
+	await test.step('Asserts the default state of selectable image cards', async () => {
+		await expect(
+			selectableImageCardsSection
+				.locator('[data-qa-id="image-card-icon-block"]')
+				.getByRole('checkbox')
+		).not.toBeChecked();
+	});
+
+	await test.step('Card becomes active when checkbox is checked', async () => {
+		for (const card of await selectableImageCards.all()) {
+			const checkbox = card.getByRole('checkbox');
+
+			await checkbox.check();
+
+			await expect(card).toHaveClass(/active/);
+
+			await checkbox.uncheck();
+
+			await expect(card).not.toHaveClass(/active/);
+		}
+	});
+
+	await test.step('Click on card title navigates to href URL', async () => {
+		await expect(page).not.toHaveURL(/#image-card-href$/);
+
+		await selectableImageCards
```

Comment:
Instead of this, you can do `await page.getByRole('link', {name: 'Beetle'}).click()`.